src/handlers/motions/motionStaked/helpers.ts

```diff
@@ -0,0 +1,209 @@
+  const unstakedSide = getMotionSide(invertedVote);
+  return {
+    address: staker,
+    stakes: {
+      raw: {
+        [stakedSide]: amount.toString(),
+        [unstakedSide]: '0',
+      },
+
+      percentage: {
+        [stakedSide]: getStakePercentage(amount, requiredStake).toString(),
+        [unstakedSide]: '0',
+      },
+    }, // TS doesn't like the computed keys, but we know they're correct
+  } as unknown as UserStakes;
+};
+
+/**
+ * Given staking data, add stakes to existing stakes and return new, updated UserStakes object
+ */
+const getUpdatedUserStakes = (
+  existingUserStakes: UserStakes,
+  amount: BigNumber,
+  vote: BigNumber,
+  requiredStake: BigNumber,
+): UserStakes => {
+  const stakedSide = getMotionSide(vote);
+  const existingRawStake = existingUserStakes.stakes.raw[stakedSide];
+  const updatedRawStake = amount.add(existingRawStake);
+  const updatedPercentage = getStakePercentage(updatedRawStake, requiredStake);
+  const updatedUserStakes: UserStakes = {
+    ...existingUserStakes,
+    stakes: {
+      raw: {
+        ...existingUserStakes.stakes.raw,
+        [stakedSide]: updatedRawStake.toString(),
+      },
+      percentage: {
+        ...existingUserStakes.stakes.percentage,
+        [stakedSide]: updatedPercentage.toString(),
+      },
+    },
+  };
+
+  return updatedUserStakes;
+};
+
+type UserStakesMapFn = (usersStakes: UserStakes) => UserStakes;
+
+const getUserStakesMapFn =
+  (
+    staker: string,
+    vote: BigNumber,
+    amount: BigNumber,
+    requiredStake: BigNumber,
+  ): UserStakesMapFn =>
+  (userStakes: UserStakes): UserStakes => {
+    const { address } = userStakes;
+
+    if (address === staker) {
+      return getUpdatedUserStakes(userStakes, amount, vote, requiredStake);
+    }
+
+    return userStakes;
+  };
+
+/**
+ * Takes existing usersStakes and updates the entry with the latest stake data
+ */
+export const getUpdatedUsersStakes = (
+  usersStakes: UserStakes[],
+  staker: string,
+  vote: BigNumber,
+  amount: BigNumber,
+  requiredStake: BigNumber,
+): UserStakes[] => {
+  const isExistingStaker = usersStakes.some(
+    ({ address }) => address === staker,
+  );
+
+  if (isExistingStaker) {
+    const updateUserStakes = getUserStakesMapFn(
+      staker,
+      vote,
+      amount,
+      requiredStake,
+    );
+
+    return usersStakes.map(updateUserStakes);
+  }
+
+  const newUserStakes = getNewUserStakes(staker, vote, amount, requiredStake);
+
+  return [...usersStakes, newUserStakes];
+};
+
+export const getUserMinStake = (
+  totalStakeFraction: BigNumber,
+  userMinStakeFraction: BigNumber,
+  skillRep: BigNumber,
+): string => {
+  /*
+   * Both totalStakeFraction and userMinStakeFraction both divide by 10 to the power of 18.
+   * Since we've multiplied by both, we need to divide by 10 to the power of 18, twice.
+   */
+
+  return skillRep
+    .mul(totalStakeFraction)
+    .mul(userMinStakeFraction)
+    .div(BigNumber.from(10).pow(36))
```

Comment:
I think I even mentioned this first, when this was originally implemented in the Dapp. This can also deal with tokes that don't have 18 points of decimals, so hard-coding 36 here is not correct.

Please confirm this with Alex though, since he's the original source for this piece of logic.

Reply:
Confirmed with Alex that token decimals are useful for us when displaying token data to a user but don't affect the way get required stake or min user stake are calculated